src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1523,13 +1539,42 @@ def __init__(self,
     def get_port_start(self) -> List[int]:
         return [5500, 4420, 8009]
 
+    @staticmethod
+    def verify_non_negative_int(field, field_name):
+        if field is not None:
+            if not isinstance(field, int) or isinstance(field, bool):
+                raise SpecValidationError(f"{field_name} must be an integer")
+            if field < 0:
+                raise SpecValidationError(f"{field_name} can't be negative")
+
+    @staticmethod
+    def verify_positive_int(field, field_name):
+        NvmeofServiceSpec.verify_non_negative_int(field, field_name)
+        if field is not None and field <= 0:
+            raise SpecValidationError(f"{field_name} must be greater than zero")
+
+    @staticmethod
+    def verify_non_negative_number(field, field_name):
+        if field is not None:
+            if not isinstance(field, Number) or isinstance(field, bool):
+                raise SpecValidationError(f"{field_name} must be a number")
+            if field < 0.0:
+                raise SpecValidationError(f"{field_name} can't be negative")
+
+    @staticmethod
+    def verify_boolean(field, field_name):
+        if field is not None:
+            if not isinstance(field, bool):
+                raise SpecValidationError(f"{field_name} must be a boolean")
```

Comment:
I don't feel like these functions ought to be tied to the NVMEoF class. We could just have them defined at top level, or imported from some separate utils file (a `utils.py` already exists under the same directory as as `service_spec.py`) which would also allow other spec classes that have similar validation needs to use them without needing to reference the NVMEoF spec class.